### DISCUSSION TITLE
postgresql float8 is handle by Scalar_converter and return excetion

### DISCUSSION THREAD

**Discussion by rccc** [spawnia/sailor#60] [Q&A]
Hello,

I hope i will not waste time of anyone. I do not think it is a bug, i think i missed something in my side.

I noticed that the fields defined as float8 in postgres are handle by ScalarConverter, and then ScalarConverter return exception as my value is float/decimal but not string.

Just to realize what happens i modify ScalaConverter like this:

```
    public function fromGraphQL($value): string
    {
        if (! is_string($value) && (!is_float($value)) ) {
            $value = (string) $value;
            // throw new \InvalidArgumentException('Expected string, got ' . gettype($value));
        }

        return $value;
    }
```
And then it is ok, (value in this case is 100, not 100.00).

Many thanks in advance for reading.

Eric

**Comment by spawnia** [comment]
I can't follow. This is a client library - what does postgres have to do with it? Converting this to a discussion.

**Comment by rccc** [comment]
Dear Spawnia,

Thanks for response. 

Let me some time to look again this week end before marking as answer.

I wish a nice day.

ps: It is a pleasure to use your library. I have just write a service that parse the DocComment of the generated classes to automatically cast the data receveid from HTTP. Works like a charm, do not sure this "issue" is still relevant then.

**Comment by rccc** [comment]
Hello,

I think the problem  concerns how Hasura manage "float8" type from postgres.

I do not understand yet why some float values are handle by ScalarConverter but you are right, thislib may not be related to my question.

Many thanks for attention

Eric
